spec/models/assignment_spec.rb

```diff
@@ -377,6 +377,101 @@
     end
   end
 
+  describe '#add_group_api' do
+    let(:course) { create(:course) }
+    let(:assignment) { create(:assignment, course: course) }
+    let(:user) { create(:end_user) }
+    let(:student) { create(:student, user: user, course: course) }
+    let(:user2) { create(:end_user) }
+    let(:student2) { create(:student, user: user2, course: course) }
+
+    shared_examples 'group persistence and members' do |expected_member_count: 0|
+      it 'creates persistent group and grouping' do
```

Comment:
I would split this into two assertions, one for the first assertion and one for the third assertion (the second is just testing an association so I don't think it's necessary). Also for the third assertion I would use `find_by` rather than `where`